store/types/store.go

```diff
@@ -320,6 +320,9 @@ type KVStoreKey struct {
 // NewKVStoreKey returns a new pointer to a KVStoreKey.
 // Use a pointer so keys don't collide.
 func NewKVStoreKey(name string) *KVStoreKey {
+	if name == "" {
```

Comment:
```suggestion
	if strings.TrimSpace(name) == "" {
```

Reply:
I think preventing empty strings makes sense, but it there a reason we'd want to disallow empty spaces? If an external module wants to store keys under `"     "`, so be it.

Reply:
It might be safer to just not allow this IMHO.

Reply:
Unless you have strong objections @amaurymartiny, I'd prefer to go with @fedekunze approach.

Reply:
Fine by me!